generator/cmd/protoc-gen-gclient/testdata/rust/golden/lib.rs

```diff
@@ -0,0 +1,97 @@
+use std::sync::Arc;
+
+pub mod model;
+
+#[derive(Clone, Debug)]
+pub struct Client {
+    inner: Arc<ClientRef>,
+}
+
+#[derive(Debug)]
+struct ClientRef {
+    http_client: reqwest::Client,
+    token: String,
+}
+
+impl Client {
+    pub fn new(tok: String) -> Self {
+        let client = reqwest::Client::builder().build().unwrap();
+        let inner = ClientRef {
+            http_client: client,
+            token: tok,
+        };
+        Self {
+            inner: Arc::new(inner),
+        }
+    }
+
+    /// Secret Manager Service
+    /// 
+    ///  Manages secrets and operations using those secrets. Implements a REST
+    ///  model with the following objects:
+    /// 
+    ///  * [Secret][google.cloud.secretmanager.v1.Secret]
+    ///  * [SecretVersion][google.cloud.secretmanager.v1.SecretVersion]
```

Comment:
No action required, but we will need to convert this type of comment into something rustdoc can use to create links.

Reply:
added a todo.